components/raftstore/src/store/msg.rs

```diff
@@ -839,7 +839,8 @@ impl<EK: KvEngine> fmt::Debug for PeerMsg<EK> {
 
 impl<EK: KvEngine> PeerMsg<EK> {
     pub fn discriminant(&self) -> u8 {
-        unsafe { *(self as *const Self as *const u8) }
+        // Ref: https://doc.rust-lang.org/std/mem/fn.discriminant.html#accessing-the-numeric-value-of-the-discriminant
+        unsafe { *<*const _>::from(self).cast::<u8>() }
```

Comment:
Honestly, I prefer pattern matching, as this unsafe code relays on its layout to be `repr(u8)` which is less known and maybe error-prone.